samcli/local/lambda_service/service.py

```diff
@@ -0,0 +1,128 @@
+        host str
+            Optional. host to start the service on Defaults to '127.0.0.1'
+        stderr io.BaseIO
+            Optional stream where the stderr from Docker container should be written to
+        """
+        self.function_name_list = function_name_list
+        super(LocalLambdaService, self).__init__(lambda_runner, port=port, host=host, stderr=stderr)
+
+    def create(self):
+        """
+        Creates a Flask Application that can be started.
+        """
+        self._app = Flask(__name__)
+
+        for function_name in self.function_name_list:
+            path = '/2015-03-31/functions/{}/invocations'.format(function_name)
+            self._app.add_url_rule(path,
+                                   endpoint=path,
+                                   view_func=self._invoke_request_handler,
+                                   methods=['POST'],
+                                   provide_automatic_options=False)
+
+        self._construct_error_handling()
+
+    def run(self):
+        """
+        This starts up the (threaded) Local Server.
+        Note: This is a **blocking call**
+
+        Raises
+        ------
+        RuntimeError
+            if the service was not created
+        """
+        if not self._app:
+            raise RuntimeError("The application must be created before running")
+
+        # Flask can operate as a single threaded server (which is default) and a multi-threaded server which is
+        # more for development. When the Lambda container is going to be debugged, then it does not make sense
+        # to turn on multi-threading because customers can realistically attach only one container at a time to
+        # the debugger. Keeping this single threaded also enables the Lambda Runner to handle Ctrl+C in order to
+        # kill the container gracefully (Ctrl+C can be handled only by the main thread)
+        multi_threaded = not self.lambda_runner.is_debugging()
+
+        LOG.debug("Local Lambda Server starting up. Multi-threading = %s", multi_threaded)
+
+        # This environ signifies we are running a main function for Flask. This is true, since we are using it within
+        # our cli and not on a production server.
+        os.environ['WERKZEUG_RUN_MAIN'] = 'true'
+
+        self._app.run(threaded=multi_threaded, host=self.host, port=self.port)
+
+    def _construct_error_handling(self):
+        """
+        Updates the Flask app with Error Handlers for different Error Codes
+
+        """
+        pass
+
+    def _invoke_request_handler(self):
+        """
+        Request Handler for the Local Lambda Invoke path. This method is responsible for understanding the incoming
+        request and invoking the Local Lambda Function
+
+        Returns
+        -------
+        A Flask Response response object as if it was returned from Lambda
+
+        """
+        flask_request = request
+
+        function_name_regex = re.compile(r'/2015-03-31/functions/(.*)/invocations')
```

Comment:
you don't need a regex here. Flask should have already parsed the `{}` part out from the path.

Reply:
I constructed every endpoint but instead, I can move to letting flask do this. It will clean up a lot of this code, which is great. Will update.